server/controllers/mealsController.js

```diff
@@ -0,0 +1,87 @@
+const connection = require("../dbConnection");
+const mealSchema = require("../models/meal");
+
+const Meal = connection.model("Meal", mealSchema);
+
+const findOneWithId = async (id) => {
+    const { _doc } = await Meal.findById(id);
+    return _doc;
+};
+
+// create
+const create = async ({
+    title,
+    picURL,
+    price,
+    servingSize,
+    servingType,
+    cuisineType,
+    ingredients,
+    requirements,
+    createdAt,
+    updatedAt,
+    chefId,
```

Comment:
Not all of these are required fields, so we want the ability to create a meal without having all fields filled out

Reply:
Agree, I have updated the code to use plain old js object to use only the arguments that were passed in.

```js
    const pojsoMeal = {};
    if (title) pojsoMeal.title = title;
    if (picURL) pojsoMeal.picURL = picURL;
    if (price) pojsoMeal.price = price;
    if (servingSize) pojsoMeal.servingSize = servingSize;
    if (servingType) pojsoMeal.servingType = servingType;
    if (cuisineType) pojsoMeal.cuisineType = cuisineType;
    if (ingredients) pojsoMeal.ingredients = ingredients;
    if (requirements) pojsoMeal.requirements = requirements;
    if (createdAt) pojsoMeal.createdAt = createdAt;
    if (updatedAt) pojsoMeal.updatedAt = updatedAt;
    if (chefId) pojsoMeal.chefId = chefId;

    const { _doc } = (await Meal.create(pojsoMeal)) || {};
```